src/configuration.ts

```diff
@@ -352,6 +458,34 @@ export class ConfigurationManager implements vscode.Disposable
 
     private async addVcpkgToolchain(vcpkgRoot : string)
     {
+        let originToolchain = this.getCMakeVcpkgToolchain();
+        if (originToolchain !== undefined)
+        {
+            if (fs.existsSync(originToolchain))
+            {
+                // check whether the vcpkg path in toolchain is not the same with the path in settings
+                if (this.convertToAbsolutePath(originToolchain) !== this.convertToAbsolutePath(vcpkgRoot + '/scripts/buildsystems/vcpkg.cmake'))
+                {
+                    let result = await vscode.window.showQuickPick(['Disable', 'override vcpkg toolchain'], {canPickMany: false});
+
+                    if (result === 'Disable')
+                    {
+                        vscode.window.showInformationMessage('Vcpkg will be disabled.');
+                        this.disableVcpkg();
```

Comment:
This may break our intended execution flow. We should terminate the process here.